composite/defi-pulse/src/index-allocations/index.ts

```diff
@@ -32,13 +29,17 @@ export const getAllocations = async (
   contractAddress: string,
   setAddress: string,
   rpcUrl: string,
-): Promise<Allocations> => {
+  network: string,
+): Promise<types.TokenAllocations> => {
   const provider = new ethers.providers.JsonRpcProvider(rpcUrl)
   const index = new ethers.Contract(contractAddress, ABI, provider)
-  const info = await index.getAllocations(setAddress)
 
-  return {
-    components: info[0],
-    units: info[1],
-  }
+  const [addresses, balances] = await index.getAllocations(setAddress)
+
+  return await Promise.all(
+    addresses.map(async (address: string, i: number) => ({
+      balance: balances[i],
+      symbol: await getSymbol(address, rpcUrl, network),
```

Comment:
It would be nice to explicitly add a `decimals: 18` property here, and add a comment that `getAllocations` fn normalizes all token balances to 18 decimals.